src/Tests/InstrEngineTests/TestAppRunner/Program.cs

```diff
@@ -23,12 +25,10 @@ static int Main(string[] args)
             }
 
             string assemblyPath = args[0];
+            _resolutionBasePath = Path.GetDirectoryName(assemblyPath);
 
-            using FileStream assemblyStream = new FileStream(assemblyPath, FileMode.Open, FileAccess.Read);
-            using BinaryReader assemblyReader = new BinaryReader(assemblyStream);
-            byte[] assemblyData = assemblyReader.ReadBytes((int)assemblyStream.Length);
-
-            Assembly assembly = AppDomain.CurrentDomain.Load(assemblyData);
+            Assembly assembly = LoadAssemblyFromDisk(assemblyPath);
+            AppDomain.CurrentDomain.AssemblyResolve += OnAssemblyResolve;
```

Comment:
>AppDomain.CurrentDomain.AssemblyResolve += OnAssemblyResolve [](http://example.com/codeflow?start=12&length=60)

Did you try Assembly.LoadFrom() instead, to see what it does?

Reply:
Also, you might be able to set AppDomain.CurrentDomain.PrivateBinPath to _resolutionPath for the assembly loader to automatically load the assemblies.

Reply:
I did experiment with it a little bit, but I wasn't able to get it to work. It's possible that was an error on my part, but I didn't think it was worth investing too much time into it since we have a solution that works.

Reply:
I would really like to know if setting PrivateBinPath would work as well. That way, we wouldn't have to resort to loading the file directly.